### PR TITLE
Make user modding history public

### LINKED ISSUE #2675: "beatmapset activities" page should be visible for Beatmap Nominators
referring to this https://osu.ppy.sh/users/896613/beatmapset-activities

Right now it is apparently only visible for QAT (and probably GMT). 
Making it visible for BN would make sense since it helps tracking things like suggestions that received kudosu, which bn are able to deny in case they aren't justified etc.

### PR DESCRIPTION
**Pull request by notbakaneko**
fixes #2675 

- is terrible 🔥 
- ~~moves the user modding history stuff to `/users/{user}/modding`~~ Adds user-specific modding history stuff to `/users/{user}/modding`
- adds a little icon next to the username in discussion posts that goes to the user's modding history 
- the views, styles, etc. need to be fixed separately

### CODE REVIEW

**Comment by nanaya** [comment]
Are we removing global overview ?_? (as in not specific to certain user)

**Comment by notbakaneko** [comment]
I guess not

File: resources/views/beatmap_discussion_posts/_item.blade.php [hunk 1]
```diff
@@ -36,16 +36,8 @@
...
-                            <div class="beatmap-discussion-post__avatar">
-                                <a href="{{ route('users.beatmapset-activities', $post->user) }}">
-                                    <div class="avatar avatar--full-rounded" style="background-image: url('{{$post->user->user_avatar}}');"></div>
-                                </a>
-                            </div>
-                            <div class="beatmap-discussion-post__user">
-                                <span class="beatmap-discussion-post__user-text u-ellipsis-overflow">{!! link_to_user($post->user) !!}</span>
-                            </div>
-                        </div>
+                        @component('beatmapset_activities._user', ['user' => $post->user])
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply resources/views/beatmap_discussion_posts/_item.blade.php]: @include with no `$slot` explodes

[the remark this answers is not in the document]
Comment (nanaya) [reply resources/views/beatmap_discussion_posts/_item.blade.php]: `$slot ?? ''`?

[the remark this answers is not in the document]
Comment (notbakaneko) [reply resources/views/beatmap_discussion_posts/_item.blade.php]: just so `@include` can be used? yeah, no

[the remark this answers is not in the document]
Comment (nanaya) [reply resources/views/beatmap_discussion_posts/_item.blade.php]: but they're same thing with just with some `$slot` magic.

File: resources/assets/coffee/react/beatmap-discussions/post.coffee [hunk 2]
```diff
@@ -79,20 +79,31 @@ class BeatmapDiscussions.Post extends React.PureComponent
...
       div
         className: "#{bn}__content"
-        a
+        div
           className: "#{bn}__user-container"
-          href: laroute.route('users.show', user: @props.user.id)
           style:
             color: userColor
+
+          a
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply resources/assets/coffee/react/beatmap-discussions/post.coffee]: ┐(ツ)┌

File: resources/assets/coffee/react/beatmap-discussions/post.coffee [hunk 3]
```diff
@@ -79,20 +79,31 @@ class BeatmapDiscussions.Post extends React.PureComponent
...
+            div
+              className: "#{bn}__user-row"
+              span
+                className: "#{bn}__user-text u-ellipsis-overflow"
+                style:
+                  color: userColor
+                @props.user.username
+
+              a
+                className: "#{bn}__user-modding-history-link"
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply resources/assets/coffee/react/beatmap-discussions/post.coffee]: apparently

**Pull request merged**